### PR TITLE
Adding support for the regex mask

### PR DESCRIPTION
**Pull request by ankitgo**

### CODE REVIEW

File: src/agent/common.ts [hunk 1]
```diff
@@ -453,24 +452,23 @@ function createMaskFunction(jobEnvironment: agentifm.JobEnvironment): Replacemen
...
-                                index = input.indexOf(valueToReplace, index + 1);
-                            }
-                            return results;
-                        });
+            else if (maskHint.type === agentifm.MaskType.Regex) {
+                indexFunctions.push((input: string) => {
+                    var results: ReplacementPosition[] = [];
+                    var actualIndex = 0;
+                    while (input.length > 0) {
+                        var match = input.match(maskHint.value);
```
Comment (GitHubSriramB) [inline src/agent/common.ts]: Instead of operating on the input directly, can we use an index to operate on the input instead?

Comment (GitHubSriramB) [inline src/agent/common.ts]: input.match() does a regex match internally?

File: src/agent/common.ts [hunk 2]
```diff
@@ -453,24 +452,23 @@ function createMaskFunction(jobEnvironment: agentifm.JobEnvironment): Replacemen
...
+                        if (match === null) {
+                            break;
+                        }
+                        else {
+                            var matchString = match.toString();
+                            results.push({ start: actualIndex + match.index, length: matchString.length });
```
Comment (GitHubSriramB) [inline src/agent/common.ts]: can we avoid matchString variable & instead just use match.length?

**Comment by GitHubSriramB** [comment]
Looks good

File: src/agent/common.ts (hunk 1 above)
Comment (scottdallamura) [inline src/agent/common.ts]: There's a check somewhere else to make sure maskHint.value is not an empty string, right?

File: src/agent/common.ts [hunk 3]
```diff
@@ -523,7 +521,7 @@ function createMaskFunction(jobEnvironment: agentifm.JobEnvironment): Replacemen
             return charArray.join("");
         };
     }
-    else if (endpoints.length === 0 && maskHints.length === 0) {
+    else if (maskHints.length === 0) {
```
Comment (scottdallamura) [inline src/agent/common.ts]: It looks like this condition is unnecessary :)

File: src/agent/common.ts (hunk 2 above)
Comment (ankitgo) [inline src/agent/common.ts]: is required because match.length is always 1

File: src/agent/common.ts (hunk 1 above)
Comment (ankitgo) [inline src/agent/common.ts]: Sriram: will take your comment. And yes, inut.match does a regex match

Scott: True, we have this check on the service

File: src/agent/common.ts (hunk 3 above)
Comment (ankitgo) [inline src/agent/common.ts]: fair..Thats called merciless code reviewing.. :D :D
Fixed..

**Pull request merged**